pilz_control/include/pilz_control/pilz_joint_trajectory_controller_impl.h

```diff
@@ -146,28 +145,16 @@ template <class SegmentImpl, class HardwareInterface>
 bool PilzJointTrajectoryController<SegmentImpl, HardwareInterface>::
 handleUnHoldRequest(std_srvs::TriggerRequest&, std_srvs::TriggerResponse& response)
```

Comment:
I find the different behavior of the two service-functions `handleHoldRequest()` and `handleUnHoldRequest()` very confusing. `handleHoldRequest()` does wait until the stop is finished. In other words, the service is blocking. `handleUnHoldRequest()`, however, does not wait until the `unhold` mode is reached but returns immediately. In other words, the service is a non-blocking service. From a user perspective this is very confusing and not intuitive.

Reply:
For me the fact that `handleUnHoldRequest()` does not longer wait is a major change. Therefore, this has to be tested at the real system.

Reply:
I can see that it is confusing from this perspective. However this is what I would expect from a Hold/Unhold function: (forget our hold-/unhold requests for a moment)
- "Hold":
  - can be triggered at any time,
  - if the robot is moving, the hold process takes up time until the robot is stopped.
- "Unhold":
  - can not be triggered when the robot performs a stop motion,
  - if it can be triggered, it has an immediate effect (no need to wait for the robot).

What do you think about that?

In other words:
"If I cannot execute "Unhold" currently, the service takes longer ..."
or
"If I cannot execute "Unhold" currently, try next time ..."

Reply:
To be honest, I think the connection type is wrong. The `hold` function should be provided via [action](http://wiki.ros.org/actionlib). If an action is used as connection type, many of the described problems would not exist and the user interface would be easier to understand and would leave the decision to wait or not, to the user.

Reply:
Let's continue the discussion: https://github.com/PilzDE/pilz_robots/issues/328